index.ts

```diff
@@ -0,0 +1,207 @@
+                    NODE_OPTIONS: "--enable-source-maps",
+                    PROJECT: pulumi.getStack(),
+                    STAGE: CONFIG.stage,
+                    RESOURCE: `${resource.name}`,
+                }
+            }
+        });
+
+        lambdaFunctionArns.push(lambdaFn.invokeArn);
+    });
+
+
+    // 3.   Setup up API gateway
+
+    const api = new aws.apigatewayv2.Api("HttpApi", {
+        protocolType: "HTTP",
+        body: pulumi
+        .all(lambdaFunctionArns)
+        .apply((functionArn) => {
+            /* Unfortunately Pulumi, to my understanding, seems to not suit perfectly OpenAPI yet.
+             * Pulumi provides a highly parallelizable infrastructure, that leads to a big trouble when it is needed
+             * to connect lambdas to the API Gateway, since it's not known when Lambdas ARNs will be available.
+             * This is a workaround to fix the issue.
+             */
+            apiroutes.forEach((resource, index) => {
+                if (resource.paths) {
+                    resource.paths.forEach(path => {
+                        if (apiDefinition.paths[path]) {
+                            Object.keys(apiDefinition.paths[path]).forEach(method => {
+                                    apiDefinition.paths[path][method]['x-amazon-apigateway-integration'] = {
+                                    ...resourceIntegrationSetting,
+                                    uri: functionArn[index]
+                                };
+                            });
+                        }
+                    });
+                }
+            });
+
+            return JSON.stringify(apiDefinition)
+        }),
+    });
+
+    // Give permission to Gateway API to execute the lambda functions
+    apiroutes.forEach(resource => {
+        const lambdaFnName = pulumi.getStack().concat('_', "dev", '_', resource.name);
+
+        const permission = new aws.lambda.Permission("permission", {
+            action: "lambda:InvokeFunction",
+            function: lambdaFnName,
+            principal: "apigateway.amazonaws.com",
+            sourceArn: pulumi.interpolate`${api.executionArn}/*`
+        });
+    });
+
+
+    // 4.   Create stage and deploy
+
+    const deploy = new aws.apigatewayv2.Deployment("commit-deploy", {
+        apiId: api.id,
+        description: "Example deployment",
+    });
+
+    const apiStage = new aws.apigatewayv2.Stage("HttpApiDefaultStage", {
+        apiId: api.id,
+        deploymentId: deploy.id,
+        name: CONFIG.stage,
+        autoDeploy: true
+    });
+
+
+    // 5.   Create DynamoDB tables
+
+    const booksTable = new aws.dynamodb.Table(CONFIG.tableName, {
+        attributes: [{
+            name: "genre-year_of_publication-index",
+            type: "S",
+        }],
+        hashKey: "genre-year_of_publication-index",
+        name: CONFIG.tableName,
+        pointInTimeRecovery: {
+            enabled: false,
+        },
+        readCapacity: 1,
```

Comment:
I would set the capacity (read an write) on demand, so that we don't pay if we don't use it (otherwise you pay fixed 1 capacity per month).